src/decorator-annotator.ts

```diff
@@ -130,6 +133,26 @@ export class DecoratorClassVisitor {
     this.propDecorators.set(name, decorators);
   }
 
+  private getValueIdentifierForType(typeSymbol: ts.Symbol): ts.Identifier|null {
```

Comment:
For lowering decorators, we need to refer to constructor types. So we start off with the identifiers that represent these types. However, we want to emit them in a value position. TypeScript does not allow to use the same ts.Identifier here. Instead, this method looks for the place where the value that is associated to the type is defined and returns that identifier instead.